### PR TITLE
Document that Either should not be used in new code

### LINKED ISSUE #1610: Update "Either" documentation to indicate that "Union" is preferred.
The [`Either` section of the user documentation](https://docs.enthought.com/traits/traits_user_manual/defining.html#either) should be updated to indicate that in new code, `Union` is preferred over `Either`.

### PR DESCRIPTION
**Pull request by mdickinson**
This PR adds notes to the user manual and API documentation to discourage users from using `Either` in new code. (`Union` should be preferred.)

Closes #1610

### CODE REVIEW

File: docs/source/traits_api_reference/trait_types.rst [hunk 1]
```diff
@@ -238,10 +238,11 @@ Traits
 .. autoclass:: ToolbarButton
    :show-inheritance:
 
-.. autoclass:: Either
+.. autoclass:: Union
```
Comment (mdickinson) [inline docs/source/traits_api_reference/trait_types.rst:241]: Deliberately swapping the order so that `Union` appears before `Either` in the docs.

There's also a drive-by fix here: `:show-inheritance:` was missing for `Union`, for no reason that I can discern.

File: docs/source/traits_user_manual/defining.rst [hunk 2]
```diff
@@ -269,7 +269,7 @@ the table.
 .. index:: PythonValue(), Range(), ReadOnly(), Regex()
 .. index:: Set() String(), This, Time()
 .. index:: ToolbarButton(), true, Tuple(), Type()
-.. index:: undefined, UUID(), ValidatedTuple(), WeakRef()
+.. index:: undefined, Union(), UUID(), ValidatedTuple(), WeakRef()
```
Comment (mdickinson) [inline docs/source/traits_user_manual/defining.rst:272]: Another drive-by fix: `Union` was missing from this list of index entries.

File: docs/source/traits_user_manual/defining.rst [hunk 3]
```diff
@@ -408,7 +408,7 @@ the table.
 | Union            | Union( *val1*\ [, *val2*, ..., *valN*,                   |
 |                  | \*\*\ *metadata*] )                                      |
 +------------------+----------------------------------------------------------+
-| UUID [4]_        | UUID( [\*\*\ *metadata*] )                               |
+| UUID             | UUID( [\*\*\ *metadata*] )                               |
```
Comment (mdickinson) [inline docs/source/traits_user_manual/defining.rst:411]: Dropped the `[4]` footnote, which talked about Python 2.5; I've re-used it for `Either`.

Comment (corranwebster) [reply docs/source/traits_user_manual/defining.rst:411]: Nitpick: footnote 4 comes before footnote 3 (attached to `Function` and `Method`) in the text.

File: docs/source/traits_user_manual/defining.rst [hunk 4]
```diff
@@ -687,7 +693,7 @@ will raise a TraitError if a value of any other type is assigned. For example::
 .. _union:
 
 Union
-::::::
+:::::
```
Comment (corranwebster) [inline docs/source/traits_user_manual/defining.rst:696]: Suggestion: `Either` is immediately above this in the text.  Perhaps  have `Union` come first?

**Review by corranwebster — APPROVED** [review]
A couple of nit-picky sorts of comments.

File: docs/source/traits_user_manual/defining.rst (hunk 3 above)
Comment (mdickinson) [reply docs/source/traits_user_manual/defining.rst:411]: Thanks; fixed. At the same time, I fixed that the current footnotes start from the third natural number, rather than the second as is more conventional.

File: docs/source/traits_user_manual/defining.rst (hunk 4 above)
Comment (mdickinson) [reply docs/source/traits_user_manual/defining.rst:646]: Done.
